tests/core/conftest.py

```diff
@@ -5,8 +5,21 @@
 
 import pytest
 
+from asyncio_run_in_process import (
+    run_in_process,
+    run_in_process_with_trio,
+)
+
 
 @pytest.fixture
 def touch_path():
     with tempfile.TemporaryDirectory() as base_dir:
         yield Path(base_dir) / "touch.txt"
+
+
+@pytest.fixture(params=('use_trio', 'use_asyncio'))
+def runner(request):
+    if request.param == 'use_trio':
+        return run_in_process_with_trio
+    else:
```

Comment:
Well, [we still have not written it down](https://github.com/ethereum/snake-charmers-tactical-manual/issues/65) but this has recently been deemed an antipattern.  We recently discovered a fixture almost exactly like this in one of our other test suites that was written roughly like this.

```python
@pytest.fixture(params=('a', 'b'))
def thing(request):
    if request == 'a':
        return thing_a
    else:
        return thing_b
```

The error is subtle and it went uncaught for a long time.  (note that it compares `request` instead of `request.param`)

We've taken the stance that the above should be written as:


```python
@pytest.fixture(params=('a', 'b'))
def thing(request):
    if request == 'a':
        return thing_a
    elif request == 'b':
        return thing_b
    else:
        raise Exception("something is wrong")
```

This ensures that when there is a subtle error it hits the exception clause rather than *implicitely* running the `else` block even though the conditional doesn't actually match.